### DISCUSSION TITLE
Conventional `clean_exit_codes`?

### DISCUSSION THREAD

**Discussion by almereyda** [wezterm/wezterm#1910] [Ideas]
In #1889 we witnessed the creation of a configuration setting [`clean_exit_codes`](https://wezfurlong.org/wezterm/config/lua/config/clean_exit_codes.html), which helps with the [`exit_behavior`](https://wezfurlong.org/wezterm/config/lua/config/exit_behavior.html) `CloseOnCleanExit`.

Here I am adding two thoughts to this conversation:

1. In there, [`@tycho-kirchner` suggested](https://github.com/tycho-kirchner)

> to consider making `clean_exit_codes = { 130 }` the default

I am a little split on this one:

On the one hand we like to argue for *convention over configuration*, which helps us to reduce configuration efforts to a bare minimum, with applying what's called *sane defaults*. This could be an argument *for* the suggestion.

On the other hand, the convention of setting a clean exit code of 130 is clearly documented and thus accessible to users, and can be set *according to the documentation*, in case some person spends time to read it.

This will most likely not affect people, who are not curious to find out what the error message means:

![Bildschirmfoto von 2022-04-24 15-48-54](https://user-images.githubusercontent.com/1645308/164979682-feea2830-c6cc-4376-880e-e64e1a19ee30.png)

But for those who are, the message could also make it easier to find the link on `exit_behaviour`, e.g. by permanently underlining it, and adding another font variant, in case it's hovered. Then, a careful reader will be able to find the notice about `clean_exit_codes`at the bottom of it much easier.

2. As much as I like an allow list for clean exit codes, I am wondering which are common exit codes, knowing they are not perfectly standardised, that make sens to be added to the array. E.g. an unsuccessful SSH connection returns `255`, which wouldn't need to block me from closing the shell.

Are there any other common exit codes known, which can most probably be ignored here?

**Comment by wez** [comment]
> the message could also make it easier to find the link on `exit_behaviour`

Yeah, I think that's reasonable. I would probably opt for something like "Click here to read more" to really spell it out.

> 2\. I am wondering which are common exit codes, knowing they are not perfectly standardised, that make sense to be added to the array. E.g. an unsuccessful SSH connection returns `255`,

There is an old-school unix concept called [sysexits](https://man.openbsd.org/sysexits) for programs that want to interoperate with `sendmail`, but it is only well-known amongst grey bearded unix hackers (such as myself) and is not widely adopted.

The main issue is that you can't know for sure that if you get exit 255 that it was a failed ssh connection. If you know that you are running the `ssh` utility in your config then it may be reasonable to add 255 along with that config, but putting it as a general exit code may result in silently closing if some other program exits with that status.
